src/modules/dashboard/components/ColonyTokenManagementDialog/ColonyTokenManagementDialog.tsx

```diff
@@ -35,31 +38,31 @@ const ColonyTokenEditDialog = ({ colonyAddress, cancel, close }: Props) => {
 
   const colonyTokens = (data && data.colony.tokens) || [];
 
-  const addToken = useCallback(
-    (newTokenAddress: Address) => {
-      const newAddresses = [
-        ...colonyTokens
-          .filter((token) => !tokenIsETH(token))
-          .map(({ address }) => address),
-        newTokenAddress,
-      ];
+  const updateTokens = useCallback(
+    (updatedAddresses: Address[]) => {
       return setColonyTokensMutation({
-        variables: { input: { colonyAddress, tokenAddresses: newAddresses } },
+        variables: {
+          input: { colonyAddress, tokenAddresses: updatedAddresses },
+        },
       });
     },
-    [colonyAddress, colonyTokens, setColonyTokensMutation],
+    [colonyAddress, setColonyTokensMutation],
   );
 
   return (
     <TokenEditDialog
       cancel={cancel}
       close={close}
       tokens={colonyTokens}
-      addTokenFn={addToken}
+      updateTokens={updateTokens}
+      tokensList={
+        process.env.NODE_ENV === 'development' ? [] : tokensList.tokens
+      }
+      nativeTokenAddress={data?.colony?.nativeTokenAddress}
```

Comment:
I mean you are already passing in the colony's tokens `tokens={colonyTokens}`. This already contains the native token

Why do you need to pass the native token separately

Reply:
how we can find out from colonyTokens array which one is native?

Reply:
Now that I've looked closer at the `colony` object, you might be right.

I thought the `tokens` prop object comes with `isNativeToken` signal, but apparently not.

You can leave it as it is.